### PR TITLE
Added toString to objects used as ha message payload

### PR DESCRIPTION
**Pull request by lutovich**
With proper toString such objects could be logged in a really nice way.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/helpers/Strings.java [hunk 1]
```diff
@@ -74,4 +76,52 @@ public static String defaultIfBlank( String str, String defaultStr )
...
+            return Arrays.toString( (char[]) o );
+        }
+        else if ( o instanceof boolean[] )
+        {
+            return Arrays.toString( (boolean[]) o );
+        }
+        else if ( o.getClass().isArray() )
+        {
+            return Arrays.deepToString( (Object[]) o );
+        }
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/helpers/Strings.java]: Perhaps do the `isArray` as an outer guard of all the other array checks, so we do fewer checks in the common case?

Comment (lutovich) [inline community/kernel/src/main/java/org/neo4j/helpers/Strings.java]: Yes, sounds right

Comment (martinfurmanski) [inline community/kernel/src/main/java/org/neo4j/helpers/Strings.java]: It looks like deepToString actually does all of this internally already, unless I am missing some finer detail.

Comment (lutovich) [inline community/kernel/src/main/java/org/neo4j/helpers/Strings.java]: Type of `Arrays#deepToString` method is like this `deepToString(Object[])` so primitive arrays can't be passed in, even with a cast to `Object[]` (which will cause ClassCastException).

**Pull request merged**